src/openzeppelin/token/erc20/library.cairo

```diff
@@ -12,349 +12,303 @@ from starkware.cairo.common.uint256 import Uint256, uint256_check, uint256_eq, u
-            assert_not_zero(recipient)
-        end
-
-        let (sender_balance: Uint256) = ERC20_balances.read(account=sender)
-        with_attr error_message("ERC20: transfer amount exceeds balance"):
-            let (new_sender_balance: Uint256) = SafeUint256.sub_le(sender_balance, amount)
-        end
-
-        ERC20_balances.write(sender, new_sender_balance)
-
-        # add to recipient
-        let (recipient_balance: Uint256) = ERC20_balances.read(account=recipient)
-        # overflow is not possible because sum is guaranteed by mint to be less than total supply
-        let (new_recipient_balance: Uint256) = SafeUint256.add(recipient_balance, amount)
-        ERC20_balances.write(recipient, new_recipient_balance)
-        Transfer.emit(sender, recipient, amount)
-        return ()
-    end
-
-    func _approve{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }(owner: felt, spender: felt, amount: Uint256):
-        with_attr error_message("ERC20: amount is not a valid Uint256"):
-            uint256_check(amount)
-        end
-
-        with_attr error_message("ERC20: cannot approve from the zero address"):
-            assert_not_zero(owner)
-        end
-
-        with_attr error_message("ERC20: cannot approve to the zero address"):
-            assert_not_zero(spender)
-        end
-
-        ERC20_allowances.write(owner, spender, amount)
-        Approval.emit(owner, spender, amount)
-        return ()
-    end
-
-    func _spend_allowance{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }(owner: felt, spender: felt, amount: Uint256):
-        alloc_locals
-        with_attr error_message("ERC20: amount is not a valid Uint256"):
-            uint256_check(amount) # almost surely not needed, might remove after confirmation
-        end
-
-        let (current_allowance: Uint256) = ERC20_allowances.read(owner, spender)
-        let (infinite:          Uint256) = uint256_not(Uint256(0, 0))
-        let (is_infinite:       felt   ) = uint256_eq(current_allowance, infinite)
-
-        if is_infinite == FALSE:
-            with_attr error_message("ERC20: insufficient allowance"):
-                let (new_allowance: Uint256) = SafeUint256.sub_le(current_allowance, amount)
-            end
-
-            _approve(owner, spender, new_allowance)
-            return ()
-        end
-        return ()
-    end
-
-end
+func ERC20_allowances(owner: felt, spender: felt) -> (allowance: Uint256) {
+}
+
+namespace ERC20 {
+    //
+    // Initializer
+    //
+
+    func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        name: felt, symbol: felt, decimals: felt
+    ) {
+        ERC20_name.write(name);
+        ERC20_symbol.write(symbol);
+        with_attr error_message("ERC20: decimals exceed 2^8") {
+            assert_lt(decimals, UINT8_MAX);
```

Comment:
shouldn't this be `assert_le`? 🤔

Reply:
Including `0`, `255` should be the max, no?

Maybe we should change `UINT8_MAX` to `255` and change this to `assert_le`?